lib/Google/Client/Files.pm

```diff
@@ -0,0 +1,204 @@
+    my $url = $self->_url("/$id/export", $params);
+    my $json = $self->_request(
+        method => 'GET',
+        url => $url
+    );
+    return $json;
+}
+
+sub generate_ids {
+    my ($self, $params) = @_;
+    my $url = $self->_url('/generateIds', $params);
+    my $json = $self->_request(
+        method => 'GET',
+        url => $url
+    );
+    return $json;
+}
+
+sub get {
+    my ($self, $id, $params) = @_;
+    confess("No ID provided") unless ($id);
+    my $url = $self->_url("/$id", $params);
+    my $json = $self->_request(
+        method => 'GET',
+        url => $url
+    );
+}
+
+sub list {
+    my ($self, $params) = @_;
+    my $url = $self->_url(undef, $params);
+    my $json = $self->_request(
+        method => 'GET',
+        url => $url
+    );
+    return $json;
+}
+
+sub update_media {
+    my ($self, $id, $params, $content) = @_;
+    confess("No ID provided") unless ($id);
+    unless ( $content && %$content ) {
+        confess("No content provided to update");
+    }
+    my $url = $self->_url("/upload/drive/v3/files/$id", $params);
+    my $json = $self->_request(
+        method => 'PATCH',
+        url => $url,
+        content => encode_json($content)
+    );
+    return $json;
+}
+
+sub update {
+    my ($self, $id, $params, $content) = @_;
+    confess("No ID provided") unless ($id);
+    unless ( $content && %$content ) {
+        confess("No content provided to update");
+    }
+    my $url = $self->_url("/$id", $params);
+    my $json = $self->_request(
+        method => 'PATCH',
+        url => $url,
+        content => encode_json($content)
+    );
+    return $json;
+}
+
+sub watch {
+    my ($self, $id, $params, $content) = @_;
+    confess("No ID provided") unless ($id);
+    my $url = $self->_url("/$id/watch", $params);
+    $content = $content ? encode_json($content) : undef;
+    my $json = $self->_request(
+        method => 'POST',
+        url => $url,
+        content => $content
+    );
+    return $json;
+}
+
+=head1 NAME
+
+Google::Client::Files
+
+=head1 DESCRIPTION
+
+A file resource client used in L<Google::Client|https://metacpan.org/pod/Google::Client> to integrate with
+Googles Files REST API.
+
+See L<https://developers.google.com/drive/v3/reference/files> for documentation.
+
+=head2 copy(Str $id, HashRef $query_params, HashRef $post_content)
+=head2 create(HashRef $query_params, HashRef $post_content)
+=head2 create_media(HashRef $query_params, HashRef $post_content)
+=head2 delete(Str $id)
+=head2 empty_trash()
+=head2 export(Str $id, HashRef $query_params)
+=head2 generate_ids(HashRef $query_params)
+=head2 get(Str $id, HashRef $query_params)
+=head2 list(HashRef $query_params)
+=head2 update(Str $id, HashRef $query_params, HashRef $post_content)
+=head2 update_media(Str $id, HashRef $query_params, HashRef $post_content)
+=head2 watch(Str $id, HashRef $query_params, HashRef $post_content)
+
+=head1 AUTHOR
+
+Ali Zia, C<< <ziali088@gmail.com> >>
+
+=head1 REPOSITORY
```

Comment:
I have to admit, my dzil setup is not at all that great, can't get it to work properly (eg push to cpan automatically, generate tags automatically etc).

There was a kind man who updated my pod on another module lol, which I think he did because a site called http://cpants.cpanauthors.org/ which seems to check this information in the main module (see PR: https://github.com/ziali088/google-api-oauth-client/pull/3)